typing/typeopt.ml

```diff
@@ -244,6 +244,14 @@ let array_kind_of_elt ~elt_sort env loc ty =
   | Unboxed_vector v -> Punboxedvectorarray v
   | Product c -> c
 
+let array_kind_of_array_type env loc ty =
```

Comment:
Can this be replaced/unified with `array_type_kind`? They do nearly the same thing

Reply:
For @ccasin

Reply:
I've removed the above function, but @ccasin should check `translprim.ml` around lines 1246 and 1260.